### ISSUE TITLE
UnhandledPromiseRejectionWarning

### ISSUE THREAD

**Issue by tobecwb** [fhemberger/imagemin-gm#8]
Using the sample scripts, I got the following error:

UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Error in file: PMI100LEI.jpg

The error appear to be random in some files. Have a folder with 300 images, every time I run, I got the same error, but on different file.

Edit: the file seens to be valid. Opened on Photoshop, GIMP, and even on imagemin (with imagemin-jpegran plugin)

**Comment by fhemberger** [comment]
Hmm, can you please change your script to

```js
// ...
imagemin(['images/*.gif'], 'build/images', { use: plugins })
    .then(() => console.log('Images optimized'))
    .catch(err => console.error(err));
```

This should give you a hint what's going on.

**Comment by tobecwb** [comment]
```
Error: Error in file: PME507LDI.jpg (now it's another file)

Stream yields empty buffer
    at Socket.<anonymous> (\node_modules\gm\lib\command.js:57:17)
    at emitNone (events.js:91:20)
    at Socket.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)

```
I changed the script to read jpg files and convert to png

```
const plugins = [
  imageminGm.resize({ width: 250, height: 250, gravity: 'Center' }),
  imageminGm.convert('png')
];

imagemin(['images/*.jpg'], 'build/images', { use: plugins })
  .then(() => console.log("fim"))
  .catch(err => console.error(err));
```

**Comment by tobecwb** [comment]
I can confirm that this error occur with jpg, gif and png files.

Started GIMP, created a new blank image (640x480), and exported as gif, png and jpg.
All formats give-me the same error

Update: tried with GraphicsMagick 32 and 64 bits (Q8 and Q16). Plataform is Windows 64.

**Comment by fhemberger** [comment]
First of all, thanks for reporting!

Which versions of Node.js, GraphicsMagic and Windows are you using exactly? So I can try and reproduce the problem.

It looks like the underlying `gm` package for Node.js is not that well maintained anymore, so I guess the problem lies there (`imagemin-gm` is just a very simple wrapper).

(I'm on vacation right now, so it can take a little while before I can look into this, sorry.)

**Comment by tobecwb** [comment]
- npm = 3.8.6;
- node = v6.11.3;
- GraphicsMagick = 1.3.26-Q8 (tried the same version for 32-bit and 64-bit);
- Windows = Windows 7 64 bits;

> (I'm on vacation right now, so it can take a little while before I can look into this, sorry.)


No problem! Thanks!

**Comment by fhemberger** [comment]
Ok, I looked into this issue again and can reproduce it, if the GraphicsMagick executables are not in the PATH (thus the underlying `gm` library can't find the binaries). Can you please check it?

**Comment by fhemberger** [comment]
I've added a new beta to GitHub, you can install it using `npm install fhemberger/imagemin-gm#2.0.0-beta` (check the new syntax!) It allows you to pass the path to the gm binaries to the constructor.

Please let me know if this fixes your issue, then I'll push the new release to npm.

**Comment by tobecwb** [comment]
I can confirm that if `gm` its defined in the PATH, everything works correctly. Trying the beta version

**Comment by tobecwb** [comment]
Tested the beta version!
Sort of worked. I think you must update the documentation or the code because of following situation:

In constructor, if I pass the path in specific way, occur the same error about `Stream yields empty buffer`.
Below a small table of constructor parameter and result:

- `new ImageminGm("C:\\Program Files (x86)\\GraphicsMagick-1.3.26-Q8")` -> **error**
- `new ImageminGm("C:\\Program Files (x86)\\GraphicsMagick-1.3.26-Q8\\")` -> **ok**
- `new ImageminGm("C:/Program Files (x86)/GraphicsMagick-1.3.26-Q8")` -> **error**
- `new ImageminGm("C:/Program Files (x86)/GraphicsMagick-1.3.26-Q8/")` -> **ok**
- `new ImageminGm("C:/Program Files (x86)/GraphicsMagick-1.3.26-Q8/gm.exe")` -> **error**
- `new ImageminGm("C:\\Program Files (x86)\\GraphicsMagick-1.3.26-Q8\\gm.exe")` -> **error**

**Comment by fhemberger** [comment]
Thanks for testing, I'll update the README and check if the path ends with a trailing (back)slash. :+1:

**Comment by fhemberger** [comment]
Thanks for your help, published `v2.0.0` to npm.

Path is normalized (so your example 1-4 will work now), passing the binary itself still results in an error (but this is documented in the README).
